### PR TITLE
I want to have invoice number generated properly

### PR DESCRIPTION
**Pull request by bartoszpietrzak1994**
Instead of generating UUID, right now invoice id's pattern looks like this:

```
y/m/sequenceNumber
```

screen shot 2018-07-11 at 13 01 58

### CODE REVIEW

File: migrations/Version20180710134758.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20180710134758 extends AbstractMigration
+{
+    public function up(Schema $schema) : void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'sqlite', 'Migration can only be executed safely on \'sqlite\'.');
```
Comment (GSadee) [inline migrations/Version20180710134758.php]: `sqlite` => `mysql`

File: migrations/Version20180710134758.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+        $this->addSql('CREATE TABLE sylius_invoicing_plugin_sequence (id INTEGER NOT NULL, idx INTEGER NOT NULL, version INTEGER DEFAULT 1 NOT NULL, PRIMARY KEY(id))');
+    }
+
+    public function down(Schema $schema) : void
+    {
+        // this down() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'sqlite', 'Migration can only be executed safely on \'sqlite\'.');
```
Comment (GSadee) [inline migrations/Version20180710134758.php]: `sqlite` => `mysql`

File: src/Generator/SequentialInvoiceIdentifierGenerator.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+use Webmozart\Assert\Assert;
+
+final class SequentialInvoiceIdentifierGenerator implements InvoiceIdentifierGenerator
+{
+    /**
+     * @var RepositoryInterface
+     */
```
Comment (GSadee) [inline src/Generator/SequentialInvoiceIdentifierGenerator.php]: Change all docblocks in this file from
```
    /**
     * @var
     */
```
to
```
    /** @var */
```

**Comment by bartoszpietrzak1994** [comment]
Since using both slashes and backslashes occurred to be problematic, replacing them with hyphens was consulted with @CoderMaggie

File: spec/Generator/SequentialInvoiceNumberGeneratorSpec.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    {
+        $this->shouldHaveType(SequentialInvoiceNumberGenerator::class);
+    }
+
+    function it_implements_invoice_identifier_generator_interface(): void
+    {
+        $this->shouldImplement(InvoiceNumberGenerator::class);
+    }
+
+    function it_generates_invoice_identifier(
```
Comment (GSadee) [inline spec/Generator/SequentialInvoiceNumberGeneratorSpec.php]: identifier => number

File: spec/Generator/SequentialInvoiceNumberGeneratorSpec.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        $sequenceRepository->findOneBy([])->willReturn($sequence);
+
+        $sequence->getVersion()->willReturn(1);
+        $sequence->getIndex()->willReturn(0);
+
+        $sequenceManager->lock($sequence, LockMode::OPTIMISTIC, 1)->shouldBeCalled();
+
+        $sequence->incrementIndex()->shouldBeCalled();
+
+        $this->generate()->shouldReturn($dateTime->format('y-m-d' . '-000000001'));
```
Comment (GSadee) [inline spec/Generator/SequentialInvoiceNumberGeneratorSpec.php]: I guess you've changed the pattern 😄

File: src/Generator/SequentialInvoiceNumberGenerator.php [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        $this->sequenceFactory = $sequenceFactory;
+        $this->sequenceManager = $sequenceManager;
+        $this->dateTimeProvider = $dateTimeProvider;
+        $this->startNumber = $startNumber;
+        $this->numberLength = $numberLength;
+    }
+
+    public function generate(): string
+    {
+        $invoiceIdentifierPrefix = $this->dateTimeProvider->__invoke()->format('y-m-d') . '-';
```
Comment (GSadee) [inline src/Generator/SequentialInvoiceNumberGenerator.php]: Is it a correct format?

File: spec/Generator/SequentialInvoiceNumberGeneratorSpec.php [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+            9
+        );
+    }
+
+    function it_is_initializable(): void
+    {
+        $this->shouldHaveType(SequentialInvoiceNumberGenerator::class);
+    }
+
+    function it_implements_invoice_identifier_generator_interface(): void
```
Comment (GSadee) [inline spec/Generator/SequentialInvoiceNumberGeneratorSpec.php]: identifier => number

File: spec/Generator/SequentialInvoiceNumberGeneratorSpec.php [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        $sequenceRepository->findOneBy([])->willReturn($sequence);
+
+        $sequence->getVersion()->willReturn(1);
+        $sequence->getIndex()->willReturn(0);
+
+        $sequenceManager->lock($sequence, LockMode::OPTIMISTIC, 1)->shouldBeCalled();
+
+        $sequence->incrementIndex()->shouldBeCalled();
+
+        $this->generate()->shouldReturn($dateTime->format('y-m' . '-000000001'));
```
Comment (GSadee) [inline spec/Generator/SequentialInvoiceNumberGeneratorSpec.php]: Shouldn't it be `y/m`?

File: src/Generator/SequentialInvoiceNumberGenerator.php [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+        $this->sequenceFactory = $sequenceFactory;
+        $this->sequenceManager = $sequenceManager;
+        $this->dateTimeProvider = $dateTimeProvider;
+        $this->startNumber = $startNumber;
+        $this->numberLength = $numberLength;
+    }
+
+    public function generate(): string
+    {
+        $invoiceIdentifierPrefix = $this->dateTimeProvider->__invoke()->format('y-m') . '-';
```
Comment (GSadee) [inline src/Generator/SequentialInvoiceNumberGenerator.php]: `y/m`?

**Comment by GSadee** [comment]
Thanks Bartek 🥄

**Pull request merged**